docker/python2/cpu/Dockerfile

```diff
@@ -1,6 +1,4 @@
 # https://store.docker.com/community/images/tensorflow/tensorflow
 FROM tensorflow/tensorflow:latest
 
-ADD . /tensorlayer
-RUN pip install /tensorlayer
-ENV PYTHONUNBUFFERED=1
+RUN pip install --upgrade tensorlayer
```

Comment:
how can we grant that the latest tensorlayer is pushed to pypi before we run docker build?

Reply:
**very good** point I didn't think about ;)

I finish the whole setup and I will address this point.

I think that we should be able to trigger CircleCI doing the job after travis is done.
If it is not possible, we have two solutions:
1. Loop Sleep and wait for the new release on PyPI.
2. Abandon CircleCI and go for everything in Travis, I decided to use a dedicated CI for Docker because otherwise the Travis build would become **very** long and it's already quite long.

Reply:
How about let's do deploying pypi in circle-ci instead of travis-ci:
https://circleci.com/blog/continuously-deploying-python-packages-to-pypi-with-circleci/
(I didn't read it in detail yet, but it should be possible)

Reply:
Actually I had a struggling on how to make docker image release.
The easiest way should be the official service provided by docker, but it doesn't work well.
So I've considered to do that in travis-ci, but I didn't want to put heavy workload on travis-ci.
Now we still have the same problem with circle-ci.

I think what I've missed was the tag condition of CI tools, having that setup properly, we should be able to only build docker image when it's necessary.

Reply:
Automatic Builds are very nice for simple stuff. Conditional builds and complex pipelines are not designed to be build that way.

I don't want to release with CircleCI because Travis is running the tests and doing everything very reliably. I just need to find a way to do the synchronisation, I believe it is possible ;)